### PR TITLE
In wait_action node change duration type from int to double

### PR DESCRIPTION
**Pull request by maksymdidukh**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo simulation |

---

## Description of contribution in a few bullet points
wait_duration port for Wait node is an integer, so if we pass something like 1.3 or 2.42 than it will automatically cast to nearest integer value, so i changed type of this port from int to double.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Review by tonynajjar — APPROVED** [review]

File: nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/wait_action.hpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ class WaitAction : public BtActionNode<nav2_msgs::action::Wait>
   {
     return providedBasicPorts(
       {
-        BT::InputPort<int>("wait_duration", 1, "Wait time")
+        BT::InputPort<double>("wait_duration", 1, "Wait time")
```
Comment (tonynajjar) [inline nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/wait_action.hpp:56]: change the default to 1.0

**Comment by tonynajjar** [comment]
I wanted to say that the documentation needs updating as well but it looks like it's correct already; there was a mismatch in the code: https://navigation.ros.org/configuration/packages/bt-plugins/actions/Wait.html

**Comment by tonynajjar** [comment]
Please also modify the test: https://github.com/ros-planning/navigation2/blob/924f167382080f3ccdd000ffc34b921cb64bcf95/nav2_behavior_tree/test/plugins/action/test_wait_action.cpp#L131

**Comment by maksymdidukh** [comment]
> Please also modify the test:
> 
> https://github.com/ros-planning/navigation2/blob/924f167382080f3ccdd000ffc34b921cb64bcf95/nav2_behavior_tree/test/plugins/action/test_wait_action.cpp#L131

done

**Comment by tonynajjar** [comment]
There are also a bunch of trees that use Wait with an int, [example](https://github.com/ros-planning/navigation2/blob/924f167382080f3ccdd000ffc34b921cb64bcf95/nav2_bt_navigator/behavior_trees/nav_to_pose_with_consistent_replanning_and_if_path_becomes_invalid.xml#L40). I know it still works like that and it's not a big deal but it's slightly cleaner imo to have them as doubles as well

**Review by tonynajjar — APPROVED** [review]

**Comment by maksymdidukh** [comment]
> There are also a bunch of trees that use Wait with an int, [example](https://github.com/ros-planning/navigation2/blob/924f167382080f3ccdd000ffc34b921cb64bcf95/nav2_bt_navigator/behavior_trees/nav_to_pose_with_consistent_replanning_and_if_path_becomes_invalid.xml#L40). I know it still works like that and it's not a big deal but it's slightly cleaner imo to have them as doubles as well

done))

**Comment by SteveMacenski** [comment]
@maksymdidukh check CI, you have linting issues: https://app.circleci.com/pipelines/github/ros-planning/navigation2/10176/workflows/cbd18739-840b-42e8-bd73-899603ec4e9d/jobs/32314/tests

Also, this will need documentation updates in the migration guide that we're changing the type of an input. Please open a Migration Guide PR as well

File: nav2_behavior_tree/plugins/action/wait_action.cpp [hunk 2]
```diff
@@ -26,16 +27,17 @@ WaitAction::WaitAction(
...
     RCLCPP_WARN(
       node_->get_logger(), "Wait duration is negative or zero "
-      "(%i). Setting to positive.", duration);
+      "(%f). Setting to positive.", duration);
     duration *= -1;
   }
-
-  goal_.time.sec = duration;
+  
+  goal_.time.sec = int(duration);
```
Comment (SteveMacenski) [inline nav2_behavior_tree/plugins/action/wait_action.cpp:39]: Don't do this. Use `from_seconds`

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by maksymdidukh** [comment]
> @maksymdidukh check CI, you have linting issues: https://app.circleci.com/pipelines/github/ros-planning/navigation2/10176/workflows/cbd18739-840b-42e8-bd73-899603ec4e9d/jobs/32314/tests
> 
> Also, this will need documentation updates in the migration guide that we're changing the type of an input. Please open a Migration Guide PR as well

here it is https://github.com/ros-planning/navigation.ros.org/pull/476

**Comment by maksymdidukh** [comment]
failing test is unrelated to my PR

**Review by SteveMacenski — APPROVED** [review]

**Comment by maksymdidukh** [comment]
@SteveMacenski, could u please backport it to iron?

**Comment by SteveMacenski** [comment]
Unfortunately no, this breaks ABI using different data types. I’d be OK fudging it for something this minor if you can show that current behavior trees with int’s set in the BT XML still work and are cast correctly. If not, we can’t break everyone’s existing BTs mid-distro

**Pull request merged**